lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -785,62 +671,29 @@ def set_zlim3d(self, bottom=None, top=None, emit=True, auto=False,
             bottom, top = bottom
         if zmin is not None:
             if bottom is not None:
-                raise TypeError('Cannot pass both `zmin` and `bottom`')
+                raise TypeError("Cannot pass both 'bottom' and 'zmin'")
             bottom = zmin
         if zmax is not None:
             if top is not None:
-                raise TypeError('Cannot pass both `zmax` and `top`')
+                raise TypeError("Cannot pass both 'top' and 'zmax'")
             top = zmax
+        return self.zaxis._set_lim(bottom, top, emit=emit, auto=auto)
 
-        self._process_unit_info([("z", (bottom, top))], convert=False)
-        bottom = self._validate_converted_limits(bottom, self.convert_zunits)
-        top = self._validate_converted_limits(top, self.convert_zunits)
-
-        old_bottom, old_top = self.get_zlim()
-        if bottom is None:
-            bottom = old_bottom
-        if top is None:
-            top = old_top
-
-        if bottom == top:
-            _api.warn_external(
-                f"Attempting to set identical bottom == top == {bottom} "
-                f"results in singular transformations; automatically "
-                f"expanding.")
-        swapped = bottom > top
-        bottom, top = self.zaxis.get_major_locator().nonsingular(bottom, top)
-        bottom, top = self.zaxis.limit_range_for_scale(bottom, top)
-        if swapped:
-            bottom, top = top, bottom
-        self.zz_viewLim.intervalx = (bottom, top)
-
-        # Mark viewlims as no longer stale without triggering an autoscale.
-        for ax in self._shared_axes["z"].get_siblings(self):
-            ax._stale_viewlims["z"] = False
-        if auto is not None:
-            self._autoscaleZon = bool(auto)
-
-        if emit:
-            self.callbacks.process('zlim_changed', self)
-            # Call all of the other y-axes that are shared with this one
-            for other in self._shared_axes["z"].get_siblings(self):
-                if other is not self:
-                    other.set_zlim(self.zz_viewLim.intervalx,
-                                   emit=False, auto=auto)
-                    if other.figure != self.figure:
-                        other.figure.canvas.draw_idle()
-        self.stale = True
-        return bottom, top
+    set_xlim3d = maxes.Axes.set_xlim
+    set_ylim3d = maxes.Axes.set_ylim
+    set_zlim3d = set_zlim
 
-    def get_xlim3d(self):
+    def get_xlim(self):
         return tuple(self.xy_viewLim.intervalx)
-    get_xlim3d.__doc__ = maxes.Axes.get_xlim.__doc__
 
-    def get_ylim3d(self):
+    get_xlim.__doc__ = maxes.Axes.get_xlim.__doc__
```

Comment:
Why do you need this? Isn't the docstring inherited from `Axes` anyway?

Reply:
good catch, this became unneeded now that xlim3d is an alias of xlim rather than the other way round.